src/AccessLogEntry.cc

```diff
@@ -225,3 +225,26 @@ AccessLogEntry::packReplyHeaders(MemBuf &mb) const
         reply->packHeadersUsingFastPacker(mb);
 }
 
+struct timeval
+AccessLogEntry::CacheDetails::trTime(const RecordTime &endTime) const
+{
+    struct timeval result;
+    memset(&result, 0, sizeof(result));
+
+    if (start_time.tv_sec)
+        tvSub(result, start_time, endTime.legacySystemTime);
```

Comment:
I checked that Squid will log a negative value (e.g., %tr=-331) if for some reason legacySystemTime<start_time.

Reply:
Thank you for that check! Out-of-order system_clock values should be rare on production servers, but they are not impossible. We will need to address that problem (possibly in a dedicated PR and probably _not_ in this PR).